### DISCUSSION TITLE
Problem in execution through terminal

### DISCUSSION THREAD

**Discussion by Rajalakshmi-Palaniappan** [JacobBumgarner/VesselVio#57] [Q&A] [answered]
Hi Jacob,

Thanks for vesselvio. I see that the application is directly downloadable for windows and Mac. I use Linux, so I tried to use it through terminal. I downloaded the zipped code from github and installed the dependencies as in the requirements file. I tried it the with the dataset under the volumes folder (JIT_volume).  when I executed the VVTerminal.py the following error shows up

TypeError: 'NoneType' object is not iterable

Here is the Traceback:
Processing dataset: JIT_volume
Traceback (most recent call last):
  File "/home/rajalakshmi/Segmentation/VesselVio-main/VVTerminal.py", line 465, in <module>
    process_volume(compiler_file, gen_options, no_anno, vis_options, 0, verbose)
  File "/home/rajalakshmi/Segmentation/VesselVio-main/VVTerminal.py", line 137, in process_volume
    annotation_data = tree_processing.convert_annotation_data(
  File "/home/rajalakshmi/Segmentation/VesselVio-main/library/annotation/tree_processing.py", line 172, in convert_annotation_data
    for region in regions:
TypeError: 'NoneType' object is not iterable

**Comment by JacobBumgarner** [comment, marked as answer]
Thanks for pointing out this bug!!

I've identified the problem and went ahead and pushed a hot fix to **main**. 

Would you be willing to download the **main** source code again and give it another try?

**Reply by JacobBumgarner** [reply]
The bug was an issue with the ordering of processing annotation data.

Specifically, I changed:
```python
    annotation_data = tree_processing.convert_annotation_data(
        ann_options.annotation_regions, ann_options.annotation_atlas
    )
    if ann_options.annotation_type == "None":
        annotation_data = {None: None}
    else:
        roi_array = segmentation_prep.build_roi_array(
            annotation_data, annotation_type=ann_options.annotation_type
        )
```

To this. 

```python
    if ann_options.annotation_type == "None":
        annotation_data = {None: None}
    else:
        annotation_data = tree_processing.convert_annotation_data(
            ann_options.annotation_regions, ann_options.annotation_atlas
        )
        roi_array = segmentation_prep.build_roi_array(
            annotation_data, annotation_type=ann_options.annotation_type
        )

```

 I haven't touched this VVTerminal code for a while and am planning on deprecating it, but I'm happy that you helped to point out a maintenance issue.

**Comment by Rajalakshmi-Palaniappan** [comment]
Hi Jacob,

Thanks for the very quick reply and fixing it. Now I could run it, however it was showing some file not found error for the results_folder.
I included the path in the vesselvio_main/library/results_export.py file in line 98:

line 98 os.mkdir('/home/rajalakshmi/Segmentation/vesselvio_results/results_folder/')

however it again shows the same error while writing the VesselVio 3D Dataset Analysis Results.xlsx file.

traceback:

Traceback (most recent call last):
  File "/home/rajalakshmi/Segmentation/VesselVio-main/VVTerminal.py", line 464, in <module>
    process_volume(compiler_file, gen_options, no_annotation, vis_options, 0, verbose)
  File "/home/rajalakshmi/Segmentation/VesselVio-main/VVTerminal.py", line 321, in process_volume
    ResExp.write_results(results_folder, gen_options.image_dimensions)
  File "/home/rajalakshmi/Segmentation/VesselVio-main/library/results_export.py", line 117, in write_results
    create_results_file(results_file, results)
  File "/home/rajalakshmi/Segmentation/VesselVio-main/library/results_export.py", line 84, in create_results_file
    wb.save(results_file)
  File "/home/rajalakshmi/anaconda3/lib/python3.9/site-packages/pyexcelerate/Workbook.py", line 102, in save
    with open(filename_or_filehandle, "wb") as fp:
FileNotFoundError: [Errno 2] No such file or directory: 'Results/Path/Here/VesselVio 3D Dataset Analysis Results.xlsx'

**Reply by JacobBumgarner** [reply]
So for that last line, you need to paste in the directory/folder location that you want the results to be saved in.

You modify that file location here:
https://github.com/JacobBumgarner/VesselVio/blob/1611501661d9fd74be42a829f636779d4eb36d38/VVTerminal.py#L441-L445

Specifically, you want to change this line:
https://github.com/JacobBumgarner/VesselVio/blob/1611501661d9fd74be42a829f636779d4eb36d38/VVTerminal.py#L443

**Comment by Rajalakshmi-Palaniappan** [comment]
Thanks for the reply and sorry for the back to back report:

Could not load .nii file using nibabel: No such file or no access: 'VOLUME_FILE.nii'
Unable to read image file using skimage.io.imread: No such file: '/home/rajalakshmi/Segmentation/VesselVio-main/VOLUME_FILE.nii'
Traceback (most recent call last):
  File "/home/rajalakshmi/Segmentation/VesselVio-main/VVTerminal.py", line 474, in <module>
    process_volume(file1, gen_options, no_annotation, vis_options, iteration, verbose)
  File "/home/rajalakshmi/Segmentation/VesselVio-main/VVTerminal.py", line 158, in process_volume
    volume, image_shape = ImProc.load_volume(volume_file, verbose=verbose)
TypeError: cannot unpack non-iterable NoneType object

**Reply by JacobBumgarner** [reply]
That's ok! Happy to walk you through this.

So what type of file are you trying to load? The `"VOLUME_FILE.nii"` is the placeholder for whatever image file you're trying to load.

**Comment by JacobBumgarner** [comment]
If you look here:
https://github.com/JacobBumgarner/VesselVio/tree/main/library/volumes

You'll see that there are two example volumes. If you want to visualize them, make sure you change the visualization options in the `VVTerminal.py` file.

**Comment by Rajalakshmi-Palaniappan** [comment]
Hi Jacob,

Thanks a lot for replying!
Now I could view them.

**Reply by JacobBumgarner** [reply]
Great to hear that! Feel free to reach out again if you need any more help.

Cheers
